gems/aws-sdk-core/lib/aws-sdk-core/param_validator.rb

```diff
@@ -71,9 +71,10 @@ def structure(ref, values, errors, context)
         end
 
         if @validate_required && shape.union
-          if values.length > 1
+          set_values = @input ? values.length : values.to_h.length
+          if set_values > 1
```

Comment:
What is values here? An array? This seems weird.

Reply:
During normal input validation this should be a hash.  However, in the case of stubbing, it will be a Union type.  the `length` method on that will just return the number of defined members rather than the number of members set.